src/coreclr/jit/codegencommon.cpp

```diff
@@ -6661,7 +6664,11 @@ unsigned CodeGen::getFirstArgWithStackSlot()
         // we find any non-parameters.
         assert(varDsc->lvIsParam);
 
+#ifdef TARGET_ARM
+        if (varDsc->GetStackOffset() == delta)
+#else  // UNIX_AMD64_ABI || TARGET_ARM64
         if (varDsc->GetArgReg() == REG_STK)
+#endif // TARGET_ARM
```

Comment:
Is my understanding right that this is necessary on ARM32 due to backfilling that means later parameters can be put _before_ previous args on the stack?

If this is the case I think it would be better to just find the parameter with the lowest stack offset here. Seems less fragile than this.